benchmark/common.js

```diff
@@ -88,52 +88,157 @@ Benchmark.prototype._queue = function(options) {
-      'benchmarks. Check benchmark/README.md for further instructions.');
-    process.exit(1);
-  }
+function AutocannonBenchmarker() {
+  const autocannon_exe = process.platform === 'win32'
+                         ? 'autocannon.cmd'
+                         : 'autocannon';
+  this.present = function() {
+    var result = child_process.spawnSync(autocannon_exe, ['-h']);
+    if (result.error && result.error.code === 'ENOENT')
+      return false;
+    else
+      return true;
+  };
+  this.create = function(path, duration, connections) {
+    const args = ['-d', duration, '-c', connections, '-j', '-n',
+                  'http://127.0.0.1:' + exports.PORT + path ];
+    var child = child_process.spawn(autocannon_exe, args);
+    child.stdout.setEncoding('utf8');
+    return child;
+  };
+  this.processResults = function(output) {
+    let result;
+    try {
+      result = JSON.parse(output);
+    } catch (err) {
+      // Do nothing, let next line handle this
+    }
+    if (!result || !result.requests || !result.requests.average) {
+      return undefined;
+    } else {
+      return result.requests.average;
+    }
+  };
 }
 
-// benchmark an http server.
-const WRK_REGEXP = /Requests\/sec:[ \t]+([0-9\.]+)/;
-Benchmark.prototype.http = function(urlPath, args, cb) {
-  hasWrk();
-  const self = this;
-
-  const urlFull = 'http://127.0.0.1:' + exports.PORT + urlPath;
-  args = args.concat(urlFull);
-
-  const childStart = process.hrtime();
-  const child = child_process.spawn('wrk', args);
-  child.stderr.pipe(process.stderr);
+function WrkBenchmarker() {
+  this.present = function() {
+    var result = child_process.spawnSync('wrk', ['-h']);
+    if (result.error && result.error.code === 'ENOENT')
+      return false;
+    else
+      return true;
+  };
+  this.create = function(path, duration, connections) {
+    const args = ['-d', duration, '-c', connections, '-t', 8,
+                  'http://127.0.0.1:' + exports.PORT + path ];
+    var child = child_process.spawn('wrk', args);
+    child.stdout.setEncoding('utf8');
+    child.stderr.pipe(process.stderr);
+    return child;
+  };
+  const regexp = /Requests\/sec:[ \t]+([0-9\.]+)/;
+  this.processResults = function(output) {
+    const match = output.match(regexp);
+    const result = match && +match[1];
+    if (!result)
+      return undefined;
+    else
+      return result;
+  };
+}
 
-  // Collect stdout
-  let stdout = '';
-  child.stdout.on('data', (chunk) => stdout += chunk.toString());
+const HTTPBenchmarkers = {
+  autocannon: new AutocannonBenchmarker(),
+  wrk: new WrkBenchmarker()
+};
 
-  child.once('close', function(code) {
-    const elapsed = process.hrtime(childStart);
-    if (cb) cb(code);
+// Benchmark an http server.
+Benchmark.prototype.http = function(urlPath, duration, connections, cb) {
+  const self = this;
+  duration = 1;
+
+  const picked_benchmarker = process.env.NODE_HTTP_BENCHMARKER ||
```

Comment:
I think we should prefer `wrk` and fallback to `autocannon`, just because `autocannon` depends on Node. Yes it shouldn't really matter, but its better to be safe.